### PR TITLE
[two_dimensional_scrollables] Fix pinned row painting when one axis is reversed

### PR DESCRIPTION
**Pull request by Piinks**
Fixes https://github.com/flutter/flutter/issues/136897

There are a bunch of changes out for TableView right now, so I will update the changelog/pubspec when I have a better idea of which ones are landing before this.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/two_dimensional_scrollables/lib/src/table_view/table.dart [hunk 1]
```diff
@@ -803,7 +803,7 @@ class RenderTableViewport extends RenderTwoDimensionalViewport {
           axisDirectionIsReversed(horizontalAxisDirection)
               ? 0.0
               : _pinnedColumnsExtent,
-          axisDirectionIsReversed(horizontalAxisDirection)
+          axisDirectionIsReversed(verticalAxisDirection)
```
Comment (Piinks) [inline packages/two_dimensional_scrollables/lib/src/table_view/table.dart:806]: Alas! A typo from https://github.com/flutter/packages/pull/5038 😅

**Comment by Piinks** [comment]
Version override: I have two other PRs out. Will update the version to include all of them in one release.

File: packages/two_dimensional_scrollables/lib/src/table_view/table.dart (hunk 1 above)
Comment (goderbauer) [reply packages/two_dimensional_scrollables/lib/src/table_view/table.dart:806]: Ups, and I didn't catch it in my review 😝 

Thanks for adding a test for this case!

File: packages/two_dimensional_scrollables/test/table_view/table_test.dart [hunk 2]
```diff
@@ -1365,10 +1365,11 @@ void main() {
 
     testWidgets('paint rects are correct when reversed and pinned',
         (WidgetTester tester) async {
+      // Regression test for https://github.com/flutter/flutter/issues/135386
```
Comment (goderbauer) [inline packages/two_dimensional_scrollables/test/table_view/table_test.dart:1368]: nit: want to maybe move this to line 1422 where the actual regression test for this begins to keep the context? The first few things in the test are unrelated, right?

Or maybe we should just have a file that contains all the screenshot tests as a separate test case and the whole file gets the comment that it should be deleted and migrated to mock_canvas once available? It seems like this text case is getting pretty large and I am afraid we will miss cases when we migrate the monolith over to mock_canvas...

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/two_dimensional_scrollables/test/table_view/table_test.dart (hunk 2 above)
Comment (Piinks) [reply packages/two_dimensional_scrollables/test/table_view/table_test.dart:1368]: Oh I must've missed this in the merge. I added links to each issue this test was covering for that very reason to have references for the rewrite. Let me go find them again

**Pull request merged**